lib/WebSocket.js

```diff
@@ -610,43 +613,32 @@ function initAsClient (address, protocols, options) {
 
   if (agent) requestOptions.agent = agent;
 
-  const req = httpObj.get(requestOptions);
+  this._req = httpObj.get(requestOptions);
+
+  this._req.on('error', (error) => {
+    if (this._req.aborted) return;
```

Comment:
This is to avoid "socket hang up" errors which are emitted when a request is aborted and a response is not received.

We can remove the check as the error would be swallowed anyway but I think being explicit is better here.